### DISCUSSION TITLE
issue with combining `vips_autorot` and `vips_icc_transform`

### DISCUSSION THREAD

**Discussion by richardmonette** [libvips/libvips#2908] [General]
Apologies in advance for this somewhat vague report. At a bit of an impasse trying to track this bug down and are hoping by sharing perhaps some light can be shed.

Basically, the sequence of operations is loading an AVIF image, running `vips_autorot`, then `vips_icc_transform` and saving out as png, which is resulting in a corrupted image which looks like:

![test_avif](https://user-images.githubusercontent.com/1041278/177209641-379ae488-5117-43cc-909e-4450ed50283e.png)

The issue is occurring erratically in production, which has made producing an exact set of replication steps challenging. One interesting observation is that if we skip either `vips_autorot` or `vips_icc_transform` then things seem to work (for the one operation kept on) as expected, which potentially suggests there is some interaction.

Perhaps someone might be able to suggest a direction for further debugging. Thanks!

**Comment by jcupitt** [comment]
Hi @richardmonette,

Huh that's an odd one. It looks maybe like a byte ordering issue (high and low swapped on a 16-bit image), but I can't imagine how that happened.

You shouldn't need to use autorot on AVIF images. The HEIF container format has a range of strange issues around autorot and we solve them (mostly) by relying on the codec autorotate and never passing any of the hints downstream.

**Comment by richardmonette** [comment]
Hey John, thank you for your message. That's a good tip about not needing autorot on AVIF, since the EXIF isn't passed down anyways. We'll try disabling the autorot for AVIF and see if that resolves the problem.
